### ISSUE TITLE
SPFX React webparts stopped working

### ISSUE THREAD

**Issue by stellanr** [SharePoint/sp-dev-docs#8544]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version: Edge Version 105.0.1343.53 (Official build) (64-bit)

- SPFx version: 1.13.1

- Node.js version: 14.15.0

### Describe the bug / error

We have multiple SPFX webparts in a production environment that all stopped working (as of today probably). They have been working just fine for months or years. They have different versions of SPFX and Node, but all have the same issue.  At first, I thought it was something regarding the past issues related to the SharePoint React 17 upgrade (that I experienced and solved in another tenant), so I have read most information in the issues here: https://github.com/SharePoint/sp-dev-docs/issues?q=is%3Aissue+label%3Areact-17+is%3Aclosed

This issue appears as "Something went wrong" in the UI of the webpart and the error details are attached here, as well as some of the errors visible in the dev console.
![SP_Page_Error_Information2](https://user-images.githubusercontent.com/25481579/199215144-bc1f16e5-65ac-454b-9ea8-85fe17ea76c5.png)

![Dev_console_errors2](https://user-images.githubusercontent.com/25481579/199215162-c9d99409-5e32-4ddf-b529-1529a5efa07a.png)

Is this in any way related to the earlier issues regarding the React update? I can see that my `package-lock.json` have occurrences of the  `@microsoft/office-ui-fabric-react-bundle` as required for example `@microsoft/sp-property-pane` and `@microsoft/sp-dialog`. But `@microsoft/office-ui-fabric-react-bundle` is not imported at all in any file in the solution. What I can see at the moment, the `BaseDialog` from sp-dialog is used in most of the affected webparts.

My dependencies in `package.json`:
![image](https://user-images.githubusercontent.com/25481579/199217820-ffe9ff8b-88b4-4fae-aa3a-51d9f1882ddd.png)

### Steps to reproduce

Not sure if applicable, but:
1. Create SPFX app using version 1.13.1 and NodeJS 14.15.0
2. Install and Import the BaseDialog from `@microsoft/sp-dialog`  (don´t know if this is the issue, I use the DialogContent from 'office-ui-fabric-react' as well)
3. Bundle the webpart and add to a page

### Expected behavior

The webpart should be working as normal on the SharePoint Page, using the selected versions/components. Any help regarding this issues would be appreciated.

**Comment by mdissel** [comment]
We've got the same problem on some tenants.

**Comment by sosandumsft** [comment]
Same issue on my tenant

**Comment by AJIXuMuK** [comment]
@stellanr - do you have any modifications in `config/write-manifests.json` file?

**Comment by mdissel** [comment]
> @stellanr - do you have any modifications in `config/write-manifests.json` file?

We didn't make any changes to that file.
```
{
  "$schema": "https://developer.microsoft.com/json-schemas/spfx-build/write-manifests.schema.json",
  "cdnBasePath": "<!-- PATH TO CDN -->"
}
```

**Comment by stellanr** [comment]
> @stellanr - do you have any modifications in `config/write-manifests.json` file?

No, just the default(?) content:
{
  "$schema": "https://developer.microsoft.com/json-schemas/spfx-build/write-manifests.schema.json",
  "cdnBasePath": ""
}

Made some more tests in the same tenant as well: if I provision a new SPFX webpart, using the stated versions of SPFX (1.13.1) and NodeJS (14.15.0) and directly bundle and package the solution and then deploy it to my app catalog, the same issues occur. That is, a plain vanilla webpart with no modifications after the `yo @microsoft/sharepoint@1.13.1 command.`

**Comment by stellanr** [comment]
Some more information: I have tested to deploy the same "vanilla" webpart mentioned before, as well as my initial webpart (from the 'production tenant' with the original issue)  to two other tenants, and there they function as expected.

**Comment by AJIXuMuK** [comment]
@stellanr - could you please share the tenant details (domain)?
You can do that via email: aterentiev@microsoft.com

**Comment by AJIXuMuK** [comment]
@stellanr - and could you also provide a stack trace for the failing network call?

Also - do you have a Private CDN enabled on the failing tenant?

**Comment by stellanr** [comment]
We do have private CDN enabled (for some "Organizational Asset libraries"). Email sent with other information.

**Comment by mdissel** [comment]
In our case the site that hosts the webpart is a classic site.

**Comment by siegfriedcw** [comment]
I'm facing the same issue. Appears to happen with everything SPFx. A provisioned Microsoft 365 Learning pathways solution from the SharePoint lookbook stopped working after a year without hassle, same goes for a custom-build SPFx application customizer.

@AJIXuMuK question regarding CDN config caught my eye so I went into three different tenants and compared settings, all of them with public & private CDN enabled. One is a M365 Developer sandbox, the other two are regular paid subscription based M365 tenants.

++++++++++++++++++++++++++++++++++++++++++
First Tenant that works:

Get-PnPTenantCdnPolicies -CdnType Public
IncludeFileExtensions MAP
Get-PnPTenantCdnPolicies -CdnType Private
IncludeFileExtensions CSS,EOT,GIF,ICO,JPEG,JPG,JS,PNG,SVG,TTF,WOFF,WOFF2

Get-PnPTenantCdnOrigin -CdnType Public
*/CLIENTSIDEASSETS
*/MASTERPAGE
*/STYLE LIBRARY
Get-PnPTenantCdnOrigin -CdnType Private
*/USERPHOTO.ASPX
*/SITEASSETS
SITES/BRANDING/PICTURES
SITES/BRANDING/TEMPLATES
++++++++++++++++++++++++++++++++++++++++++
Second Tenant that works (M365 Developer sandbox):

Get-PnPTenantCdnPolicies -CdnType Public
IncludeFileExtensions CSS,EOT,GIF,ICO,JPEG,JPG,JS,MAP,PNG,SVG,TTF,WOFF,WOFF2
Get-PnPTenantCdnPolicies -CdnType Private
IncludeFileExtensions GIF,ICO,JPEG,JPG,JS,PNG,GLB

Get-PnPTenantCdnOrigin -CdnType Public
*/MASTERPAGE
*/STYLE LIBRARY
*/CLIENTSIDEASSETS
Get-PnPTenantCdnOrigin -CdnType Private
*/SITEASSETS
++++++++++++++++++++++++++++++++++++++++++
Third, and failing, Tenant:

Get-PnPTenantCdnPolicies -CdnType Public
IncludeFileExtensions MAP,JS,CSS
Get-PnPTenantCdnPolicies -CdnType Private
IncludeFileExtensions EOT,GIF,ICO,TTF,WOFF,WOFF2,JPEG,JFIF,JPG,PNG,SVG

Get-PnPTenantCdnOrigin -CdnType Public
*/CLIENTSIDEASSETS
*/MASTERPAGE
*/STYLE LIBRARY
Get-PnPTenantCdnOrigin -CdnType Private
*/USERPHOTO.ASPX
SITES/BRANDING/PICTURES
SITES/BRANDING/TEMPLATES
*/SITEASSETS
SITES/CDN/STYLE LIBRARY
SITES/LEARNING/SITEASSETS
++++++++++++++++++++++++++++++++++++++++++

After re-configure the failing Tenant to use the following it immediately started working again:
Get-PnPTenantCdnPolicies -CdnType Public
IncludeFileExtensions MAP
Get-PnPTenantCdnPolicies -CdnType Private
IncludeFileExtensions CSS,EOT,GIF,ICO,JPEG,JPG,JS,PNG,SVG,TTF,WOFF,WOFF2

Might be helpful for others even though not anyone can simply change its Tenant config like I could (we are a medium size business).

Cheers,

Sig

**Comment by AJIXuMuK** [comment]
This should be fixes as of now. Please confirm.

**Comment by mdissel** [comment]
Confirmed that it's fixed! Any tips how we can prevent this in the future? was is related to a classic site?

**Comment by stellanr** [comment]
Yes, all webparts that I just checked seem to be working as expected again! But, I misread a bit and thought that you meant to confirm if the setting of allowed file extensions for the CDN (using Set-PnPTenantCdnPolicies -CdnType Private
IncludeFileExtensions) solved the issue... so I did this before I checked the webparts. So I guess that it would have worked without running that command according to @AJIXuMuK if I understand correct. Can add that all sites but one are modern Team sites or Communication sites.

Is there any information regarding what caused these issues?

Thanks for all your help on this issue!
